flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/common/utils/SchemaTransform.java

```diff
@@ -225,6 +227,26 @@ private static Schema handleAvroLogicalTypes(
                 Schema geoSchema = Schema.create(Schema.Type.STRING);
                 geoSchema.addProp(LogicalType.LOGICAL_TYPE_PROP, "geography_wkt");
                 return geoSchema;
+            case "TIME":
+                return LogicalTypes.timeMicros().addToSchema(Schema.create(Schema.Type.LONG));
+            case "DATETIME":
+                return LogicalTypes.localTimestampMicros()
+                        .addToSchema(Schema.create(Schema.Type.LONG));
+            case "DATE":
+                return LogicalTypes.date().addToSchema(Schema.create(Schema.Type.INT));
```

Comment:
It seems like Beam has a more elaborate conversion process for `time` types: [link](https://github.com/apache/beam/blob/master/sdks/java/extensions/avro/src/main/java/org/apache/beam/sdk/extensions/avro/schemas/utils/AvroJavaTimeConversions.java). How did you arrive at this condensed implementation?

Reply:
I think this is to enable support for localTimestamp, we use avro 1.11.1 which already has support for the same. Their code (as they state) is copied from  avro 1.10.2.